### ISSUE TITLE
How to set regexp flags?

### ISSUE THREAD

**Issue by radry** [huginn/huginn#3355]
I need to set the ungreedy flag U but somehow it doesn't work. There are no clear instructions on how to set a flag.

In the Event Formatter Agent there is this example, which I think means A flag? However trying to set the U flag doesn't work and the result is empty. 

`"regexp": "\A(?<time>\d\d:\d\d [AP]M [A-Z]+)",`

I tested the regexp with other tools and it works. Also, without the flag it also works but is greedy.

**Comment by Unending** [comment]
`\A` just means the start of the string.

I don't think Ruby regex supports the ungreedy flag, a lot of implementations don't, it's actually strongly discouraged.

Why not just use possessive quantifiers?

**Comment by radry** [comment]
In the meantime I managed to change the regexp in such a way it works by only matching the correct string, so Ungreedy is not needed anymore.  
Regardless it would be good to know how to correctly set flags. The instructions even mention that flags can be set, but now how it's done correctly.

**Comment by Unending** [comment]
Yeah, but that's just knowing [how regex works](https://www.regular-expressions.info/refmodifiers.html), not exactly a uniquely Huginn problem.

I'm guessing that line in the agent description was added to reduce support calls. It might be valuable to add that Huginn only supports regex mode modifiers supported by Ruby. Feel free to extend the documentation and make a PR.
